lib/Sema/SemaResolveType.cpp

```diff
@@ -483,13 +483,44 @@ void TypeResolver::visit(CallExprType &CE) {
       case Identifier::SpecialIdentifier::abi_encodePacked:
       case Identifier::SpecialIdentifier::abi_encode:
         break;
-      case Identifier::SpecialIdentifier::abi_encodeWithSelector:
+      case Identifier::SpecialIdentifier::abi_encodeWithSignature: {
+        auto &Arg0 = CE.getRawArguments().at(0);
+        if (auto *IC = dynamic_cast<ImplicitCastExpr *>(Arg0.get())) {
+          Actions.resolveImplicitCast(*IC, std::make_shared<BytesType>(),
+                                      false);
+        }
+        std::vector<ExprPtr> Keccak256Arguments;
+        Keccak256Arguments.emplace_back(std::move(Arg0));
+        auto CallKeccak256 = std::make_unique<CallExpr>(
+            SourceRange(),
+            std::move(std::make_unique<Identifier>(
+                Token(), Identifier::SpecialIdentifier::keccak256, nullptr)),
+            std::move(Keccak256Arguments));
+        CallKeccak256->setType(
+            std::make_shared<FixedBytesType>(FixedBytesType::ByteKind::B32));
+        Arg0 = Actions.CreateDummy(std::move(CallKeccak256));
+        Arg0 = Actions.CreateDummy(std::move(Arg0));
```

Comment:
Why does `Arg0` need to pass into the `CreateDummy` twice?

Reply:
Comment added